specs/merge/fork-choice.md

```diff
@@ -66,44 +67,59 @@ def finalize_block(self: ExecutionEngine, block_hash: Hash32) -> bool:
     ...
 ```
 
-## Containers
+## Helpers
 
-#### `PowBlock`
+### `TransitionStore`
 
 ```python
-class PowBlock(Container):
+@dataclass
+class TransitionStore(object):
+    transition_total_difficulty: uint256
+```
+
+### `PowBlock`
+
+```python
+@dataclass
+class PowBlock(object):
     block_hash: Hash32
     is_processed: boolean
     is_valid: boolean
     total_difficulty: uint256
 ```
 
-## Helper functions
-
-#### `get_pow_block`
+### `get_pow_block`
 
 Let `get_pow_block(block_hash: Hash32) -> PowBlock` be the function that given the hash of the PoW block returns its data.
 
 *Note*: The `eth_getBlockByHash` JSON-RPC method does not distinguish invalid blocks from blocks that haven't been processed yet. Either extending this existing method or implementing a new one is required.
 
-#### `is_valid_transition_block`
+### `get_transition_store`
+
+```python
+def get_transition_store(anchor_pow_block: PowBlock):
+    transition_total_difficulty = anchor_pow_block.total_difficulty + TRANSITION_TOTAL_DIFFICULTY_OFFSET
+    return TransitionStore(transition_total_difficulty=transition_total_difficulty)
+```
+
+### `is_valid_transition_block`
 
 Used by fork-choice handler, `on_block`.
 
 ```python
-def is_valid_transition_block(block: PowBlock) -> bool:
-    is_total_difficulty_reached = block.total_difficulty >= TRANSITION_TOTAL_DIFFICULTY
+def is_valid_transition_block(transition_store: TransitionStore, block: PowBlock) -> bool:
```

Comment:
Maybe rename to `is_valid_terminal_pow_block` or something like that? The `is_valid_transition_block` name may mislead a bit as "the transition block" also refers to the first embedded block.

Reply:
Good suggestion 👍